src/lib.rs

```diff
@@ -35,15 +39,29 @@ fn index(root: &State<String>) -> OutpackResult<config::Root> {
 }
 
 #[rocket::get("/metadata/list")]
-fn list(root: &State<String>) -> OutpackResult<Vec<location::LocationEntry>> {
+fn list_metadata(root: &State<String>) -> OutpackResult<Vec<location::LocationEntry>> {
     location::read_locations(root)
         .map_err(|e| OutpackError::new(e))
         .map(|r| OutpackSuccess::from(r))
 }
 
+#[rocket::get("/metadata/<id>/json")]
+fn get_metadata(root: &State<String>, id: String) -> Result<OutpackSuccess<serde_json::Value>, OutpackError> {
+    metadata::get_metadata(root, &id)
+        .map_err(|e| OutpackError::new(e))
+        .map(|r| OutpackSuccess::from(r))
+}
+
+#[rocket::get("/metadata/<id>/text")]
```

Comment:
I'm copying the R server here! https://github.com/mrc-ide/outpack.server/blob/main/R/api.R#L49

Reply:
Hmm, that means the R server README is out of date, I think :(

I wonder why we did this? I have a feeling it was about avoiding some weird serialisation/deserialisation fight in the client, but this feels dim already. So this implementation is ok and we just need to update the README on the R version (if we choose to keep it)

Reply:
I see a comment in the R server saying that it would be nicer to switch on accept header, and I think rocket supports this. Shall I change over to using that approach? 

Somewhat confusingly, I see no reference to these `text`/`json` route suffixes in the client though: https://github.com/mrc-ide/outpack/blob/fbf9bd8b90b7bfb11feb188c494d1c6a21601298/R/location_http.R#L46

Reply:
It's just this one with the suffix, no? https://github.com/mrc-ide/outpack/blob/fbf9bd8b90b7bfb11feb188c494d1c6a21601298/R/location_http.R#L24-L25

Reply:
Oh lol, you're right, I was looking at the wrong endpoint